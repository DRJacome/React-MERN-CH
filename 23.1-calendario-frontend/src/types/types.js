export const types = {
    uiOpenModal: '[ui] Open modal',
    uiCloseModal: '[ui] Close modal',

    setActiveEvent: '[event] Set active',
    addNewEvent: '[event] Add new',
    clearActiveEvent: '[event] clear active event',
    updateEvent: '[event] Event updated',
    deleteEvent: '[event] Event deleted',

    authCheckingFinish: '[auth] Finish checking login state',
    authStartLogin: '[auth] Start login',
    authLogin: '[auth] Login',
    authStartRegister: '[auth] Start register',
    authStartTokenRenew: '[auth] Start token renew',
    authLogout: '[auth] Logout',
}