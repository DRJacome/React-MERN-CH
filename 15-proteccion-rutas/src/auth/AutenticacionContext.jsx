import React, { createContext } from 'react'

export const AutenticacionContext = createContext()