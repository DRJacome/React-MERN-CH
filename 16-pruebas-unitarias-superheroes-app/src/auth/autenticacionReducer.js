import { types } from '../types/types';

/* Estado inicial:
  const state = {
  nombre: 'Fernando',
  logged: true
} */

/* El par치metro state est치 definido fuera del reducer y el par치metro action est치 desestructurado 
para acceder directamente al type desde el switch */
export const autenticacionReducer = (state = {}, action) => {
  switch (action.type) {
    case types.login:
      return {
        ...action.payload,
        logged: true
      }
    case types.logout:
      return {
        logged: false
      }
    default:
      return state
  }
}
