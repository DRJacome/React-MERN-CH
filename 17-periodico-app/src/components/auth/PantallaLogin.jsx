import React from 'react'
import { Link } from "react-router-dom"

export const PantallaLogin = () => {
    return (
        <>
            <h3 className="auth__title">Login</h3>
            <form>
                <input
                    type="text"
                    placeholder="email"
                    name="email"
                    className="auth__input"
                    autoComplete="off"
                />
                <input
                    type="password"
                    placeholder="Contrase침a"
                    name="password"
                    className="auth__input"
                />
                <button
                    type="submit"
                    className="btn btn-primary btn-block"
                    disabled={true}
                >
                    Iniciar sesi칩n
                </button>
                <hr />
                <div className="auth__social-networks">
                    <p>Iniciar sesi칩n con RRSS</p>
                    <div
                        className="google-btn"
                    >
                        <div className="google-icon-wrapper">
                            <img className="google-icon" src="/assets/icono-google.png" alt="bot칩n google" />
                        </div>
                        <p className="btn-text">
                            <b>Sign in with Google</b>
                        </p>
                    </div>
                </div>
                <Link to='/auth/register' className="link">Crear cuenta</Link>
            </form>
        </>
    )
}