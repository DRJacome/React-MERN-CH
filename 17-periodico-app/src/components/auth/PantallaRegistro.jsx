import React from 'react'
import { Link } from "react-router-dom"


export const PantallaRegistro = () => {
  return (
    <>
      <h3 className="auth__title">Registro</h3>
      <form>
        <input
          type="text"
          placeholder="Nombre"
          name="nombre"
          className="auth__input"
          autoComplete="off"
        />
        <input
          type="password"
          placeholder="Contraseña"
          name="password"
          className="auth__input"
        />
        <input
          type="password"
          placeholder="Confirmar contraseña"
          name="confirmarPassword"
          className="auth__input"
        />
        <button
          type="submit"
          className="btn btn-primary btn-block mb-5"
          disabled={true}
        >
          Registrarse
        </button>
        <hr />
        
          <Link to='/auth/login' className="link">¿Ya estás registrado?</Link>
      </form>
    </>
  )
}