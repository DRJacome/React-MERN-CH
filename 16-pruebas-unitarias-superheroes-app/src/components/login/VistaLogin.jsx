import React, { useContext } from 'react'
import { AutenticacionContext } from '../../auth/AutenticacionContext'
import { types } from '../../types/types'

export const VistaLogin = ({ history }) => {

  /* Autenticación de usuario. */
  const { dispatch } = useContext(AutenticacionContext)
  const gestionarLogin = () => {
    //history.push('/')

    const ultimoPath = localStorage.getItem('lastPath') || '/'

    dispatch({
      type: types.login,
      payload: {
        nombre: 'David'
      }
    })
    history.replace(ultimoPath)
  }
  return (
    <div className="container mt-5">
      <h1>¡Hola! identifícate</h1>
      <hr />
      <button className="btn btn-primary" onClick={gestionarLogin}>Identificarse</button>
    </div>
  )
}