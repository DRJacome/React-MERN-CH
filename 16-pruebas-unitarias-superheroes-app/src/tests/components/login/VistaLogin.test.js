import React from 'react'
import '@testing-library/jest-dom'
import { mount } from 'enzyme'
import { VistaLogin } from '../../../components/login/VistaLogin'
import { AutenticacionContext } from '../../../auth/AutenticacionContext'
import { types } from '../../../types/types'

describe('Pruebas en <VistaLogin/>', () => {

  const historial = {
    replace: jest.fn(),
  }

  const valorContext = {
    dispatch: jest.fn(),
    usuario: {
      logged: false
    }
  }

  const wrapper = mount(
    <AutenticacionContext.Provider value={valorContext}>
      <VistaLogin history={historial} />
    </AutenticacionContext.Provider>
  );

  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('Debe realizar el dispatch y la navegaciÃ³n', () => {

    //wrapper.find('button').prop('onClick')()
    const manejarClick = wrapper.find('button').prop('onClick')

    manejarClick();
    expect(valorContext.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload: {
        nombre: 'David'
      }
    });

    //expect(historial.replace).toHaveBeenCalled();
    expect(historial.replace).toHaveBeenCalledWith("/");

    localStorage.setItem('lastPath', '/dc')
    manejarClick();
    expect(historial.replace).toHaveBeenCalledWith("/dc");

  })
})