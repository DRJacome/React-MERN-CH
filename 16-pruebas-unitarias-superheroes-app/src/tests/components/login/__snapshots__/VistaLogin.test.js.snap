// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <VistaLogin/> Debe mostrarse correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    ¡Hola! identifícate
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Identificarse
  </button>
</div>
`;
